server/rooms.ts

```diff
@@ -305,6 +305,10 @@ export abstract class BasicRoom {
 		}
 		if (this.parent) return this.parent.getMuteTime(user);
 	}
+	getGame<T extends RoomGame>(constructor: new (...args: any[]) => T): T | null {
+		if (this.game && this.game.constructor.prototype.gameid === constructor.prototype.gameid) return this.game as T;
```

Comment:
Um, `constructor.prototype.gameid` should be `undefined` right now, I don't think you've actually defined it yet?

Reply:
```
>> room.game.constructor.prototype.gameid
<< undefined
```

Your changes don't seem to add a prototype anywhere.

Anyway, like I said, I think `static readonly gameid` is the best approach.

BUT if you wanted to be lazy, `this.game.constructor.name === constructor.name` _is_ good enough.

Reply:
I'll go with the 'lazy' route since I don't want to deal with breaking the untyped files.